src/Avalonia.Visuals/Media/IImage.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Avalonia.Platform;
+using Avalonia.Visuals.Media.Imaging;
+
+namespace Avalonia.Media
+{
+    /// <summary>
+    /// Represents a raster or vector image.
+    /// </summary>
+    public interface IImage
```

Comment:
it would be nice if `IImage ` has some event  or a way to notify parent that it is changed (force Redraw).
a `Changed` event should be enough and Image will listen for it and call `InvalidateVisual()`
Changing WriteableBitmaps like workflows (GIF?) would benefit from it so there will be no need to call `InvalidateVisual()` of the parent Image.

Reply:
Yep, you can already do that by implementing `IAffectsRender`; [`DrawingImage` implements this](https://github.com/AvaloniaUI/Avalonia/pull/3378/files#diff-f49d0afcf2e6b3330175b9d1707f2413R11) to invalidate the image when the `Drawing` changes.

Reply:
What prevents us from enforcing `IAffectsRender`?

Reply:
You mean `IImage` implementing `IAffectsRender` directly? Nothing, except that if the `IImage` implementation is immutable there's no need to implement it.

Reply:
Y I guess then it is fine to make this implementation depended.